### DISCUSSION TITLE
PC version possible or only RPI?

### DISCUSSION THREAD

**Discussion by stefan63** [Peter-van-Tol/LiteX-CNC#72] [Q&A]
Love this project thank you Peter!
**My Q's so far:**  Is it possible to install and run LiteX-CNC on a regular PC with Debian LinuxCNC ?
I tried to install it on an HP G3 mini but have loads of errors and no sucess so far.
Does a LinuxCNC iso with LiteX included exist?
I have a 5a-75e 5a-75e 8.0
Kind regards/Stefan

**Comment by Peter-van-Tol** [comment]
Yes, this should be absolutely possible. The first versions of `litexcnc` have been developed on an old `i386` with a Celeron processor (salvaged from a multi-functional).

Could you please share the errors here, so I can see where things go wrong.

**NOTE**: The following features of `litexcnc` are not supported on devices like desktop:
- SPI driver, so you must use the ethernet driver;
- flashing the firmware directly, you must use an external dongle and supply the appropriate configuration when using the command `flash_firmware`

**Reply by stefan63** [reply]
Will try again with your hints and report back :0)

**Reply by Peter-van-Tol** [reply]
@stefan63 : Any luck with the latest version?

**Comment by WrongVoltage** [comment]
Hello Peter, I´ve been trying to install and run Litex on a PC but had issues trying to build the firmware, here is the terminal info.

[litex error 1.txt](https://github.com/user-attachments/files/20742660/litex.error.1.txt)


When I install the toolchain I have issues with initramfs, the rest I didn´t notice any issues for now, but I need to write the firmware to test it!
Thanks for all your work!


PS the bashrc has the following:
```
# set PATH so it includes oss-cad-suite 
if [ -d "/home/cnc/toolchain/oss-cad-suite/bin" ] ; then 
PATH="/home/cnc/toolchain/oss-cad-suite/bin:$PATH" 
fi
```
If I run "nextpnr-ecp5" I get this message:

`nextpnr-ecp5: error while loading shared libraries: libboost_program_options.so.1.71.0: cannot open shared object file: No such file or directory`

I have 1.74.0 but can't download 1.71.0, could this be it?

PPS I have installed the LinuxCNC 2.9.4 release with debian 12.

PPPS I installed LinuxCNC 2.9.2 iso and now it works up to the flashing point, I'm using a cheap clone of the usb blaster, so I think that's the problem, will try to fix that. and add info here after testing.

**Reply by WrongVoltage** [reply]
USB BLASTER didn't work, but I believe it's not the software side of things, with FT232R I was able to program with success! Manually set the IP on the pc and was able to ping the card. I'll come back to say how things went!

**Reply by WrongVoltage** [reply]
Yep it works! Just follow [PC.RST](https://github.com/Peter-van-Tol/LiteX-CNC/blob/main/docs/src/pc.rst)
